addons/l10n_sa_edi_pos/models/pos_order.py

```diff
@@ -0,0 +1,34 @@
+from odoo import api, fields, models
+
+
+class PosOrder(models.Model):
+    _inherit = "pos.order"
+
+    l10n_sa_invoice_qr_code_str = fields.Char(related="account_move.l10n_sa_qr_code_str", string="ZATCA QR Code")
+    l10n_sa_invoice_edi_state = fields.Selection(related="account_move.edi_state", string="Electronic invoicing")
+
+    @api.model
+    def _load_pos_self_data_domain(self, data):
+        return [('id', '=', False)]
+
+    @api.model
+    def sync_from_ui(self, orders):
+        result = super().sync_from_ui(orders)
+
+        # all orders in a single sync_from_ui call will belong to the same session
+        if (
+            len(orders) > 0
+            and orders[0].get("session_id")
+            and self.env["pos.session"].browse(orders[0]["session_id"]).config_id.company_id.country_id.code == 'SA'
+        ):
+            order_ids = self.env["pos.order"].browse([order["id"] for order in result['pos.order']])
+            order_dict = {x.id: x for x in order_ids}
+
+            for order in result["pos.order"]:
+                order_db = order_dict.get(order["id"])
+                order.update({
+                    'l10n_sa_invoice_qr_code_str': order_db.l10n_sa_invoice_qr_code_str,
+                    'l10n_sa_invoice_edi_state': order_db.l10n_sa_invoice_edi_state
+                })
+
+        return result
```

Comment:
All of this isn't needed if `l10n_sa_invoice_qr_code_str` and `l10n_sa_invoice_edi_state` are in "order_db" that's means that the initial result of `sync_from_ui` already contains them